utils/python/CIME/user_mod_support.py

```diff
@@ -63,8 +63,17 @@ def apply_user_mods(caseroot, user_mods_path, ninst=None):
             case_shell_commands = shell_commands_file.replace(include_dir, caseroot)
             with open(shell_commands_file,"r") as fd:
                 new_shell_commands = fd.read().replace("xmlchange","xmlchange --force")
-            with open(case_shell_commands, "a") as fd:
+            # prepend new_shell_commands to case_shell_commands file
+            old_shell_commands = None
+            if os.path.isfile(case_shell_commands):
```

Comment:
the fd.close is implied, you don't need it.   It seems like you should build up the shell commands in memory and only write them once instead of repeatedly opening and closing the file:
shell_commands = new_shell_commands + old_shell_commands